selfdrive/car/nissan/interface.py

```diff
@@ -0,0 +1,199 @@
+    ret.longitudinalTuning.kiV = [0.]
+
+    # end from gm
+
+    # TODO: get actual value, for now starting with reasonable value for
+    # civic and scaling by mass and wheelbase
+    ret.rotationalInertia = scale_rot_inertia(ret.mass, ret.wheelbase)
+
+    # TODO: start from empirically derived lateral slip stiffness for the civic and scale by
+    # mass and CG position, so all cars will have approximately similar dyn behaviors
+    ret.tireStiffnessFront, ret.tireStiffnessRear = scale_tire_stiffness(ret.mass, ret.wheelbase, ret.centerToFront)
+
+    return ret
+
+  # returns a car.CarState
+  def update(self, c, can_strings):
+    self.pt_cp.update_strings(can_strings)
+    self.cam_cp.update_strings(can_strings)
+    self.adas_cp.update_strings(can_strings)
+
+    self.CS.update(self.pt_cp, self.adas_cp, self.cam_cp)
+
+    # create message
+    ret = car.CarState.new_message()
+
+    ret.canValid = self.pt_cp.can_valid and self.adas_cp.can_valid and self.cam_cp.can_valid
+
+    # speeds
+    ret.vEgo = self.CS.v_ego
+    ret.aEgo = self.CS.a_ego
+    ret.vEgoRaw = self.CS.v_ego_raw
+    ret.yawRate = self.VM.yaw_rate(self.CS.angle_steers * CV.DEG_TO_RAD, self.CS.v_ego)
+    ret.standstill = self.CS.standstill
+    ret.wheelSpeeds.fl = self.CS.v_wheel_fl
+    ret.wheelSpeeds.fr = self.CS.v_wheel_fr
+    ret.wheelSpeeds.rl = self.CS.v_wheel_rl
+    ret.wheelSpeeds.rr = self.CS.v_wheel_rr
+
+    # steering wheel
+    ret.steeringAngle = self.CS.angle_steers
+
+    # torque and user override. Driver awareness
+    # timer resets when the user uses the steering wheel.
+    ret.steeringPressed = self.CS.steer_override
+    ret.steeringTorque = self.CS.steer_torque_driver
+
+    ret.gas = self.CS.pedal_gas
+    ret.gasPressed = self.CS.user_gas_pressed
+
+    # brake pedal
+    ret.brakePressed = self.CS.brake_pressed != 0
+    ret.brakeLights = self.CS.brake_lights
+
+    # cruise state
+    ret.cruiseState.enabled = bool(self.CS.acc_active)
+    ret.cruiseState.speed = self.CS.v_cruise_pcm * CV.KPH_TO_MS
+    ret.cruiseState.available = bool(self.CS.main_on)
+    ret.cruiseState.speedOffset = 0.
+
+    ret.leftBlinker = self.CS.left_blinker_on
+    ret.rightBlinker = self.CS.right_blinker_on
+    ret.seatbeltUnlatched = self.CS.seatbelt_unlatched
+    ret.doorOpen = self.CS.door_open
+
+    buttonEvents = []
+
+    # blinkers
+    if self.CS.left_blinker_on != self.CS.prev_left_blinker_on:
+      be = car.CarState.ButtonEvent.new_message()
+      be.type = ButtonType.leftBlinker
+      be.pressed = self.CS.left_blinker_on
+      buttonEvents.append(be)
+
+    if self.CS.right_blinker_on != self.CS.prev_right_blinker_on:
+      be = car.CarState.ButtonEvent.new_message()
+      be.type = ButtonType.rightBlinker
+      be.pressed = self.CS.right_blinker_on
+      buttonEvents.append(be)
+
+    be = car.CarState.ButtonEvent.new_message()
+    be.type = ButtonType.accelCruise
+    buttonEvents.append(be)
+
+
+    events = []
```

Comment:
Recently made improvement in car abstraction (see other cars)
Reuse standard events by calling the proper method from the parent class `CarInterfaceBase`